tools/profcheck/check.go

```diff
@@ -0,0 +1,363 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	profiles "go.opentelemetry.io/proto/otlp/profiles/v1development"
+	"google.golang.org/protobuf/proto"
+)
+
+func CheckConformance(data *profiles.ProfilesData) error {
+	dict := data.Dictionary
+	if len(data.ResourceProfiles) == 0 {
+		return errors.New("resource profiles are empty")
+	}
+	for _, resourceProfiles := range data.ResourceProfiles {
+		// TODO: Check attributes?
+		for _, scopeProfiles := range resourceProfiles.ScopeProfiles {
+			// TODO: Check attributes?
+			for i, profile := range scopeProfiles.Profiles {
+				if err := checkProfile(profile, dict); err != nil {
+					return fmt.Errorf("profile %d: %v", i, err)
+				}
+			}
+		}
+	}
+	return checkDictionary(dict)
+}
+
+func checkProfile(prof *profiles.Profile, dict *profiles.ProfilesDictionary) error {
+	var errs error
+	if err := checkAttributeIndices(prof.AttributeIndices, dict); err != nil {
+		errs = errors.Join(errs, prefixErrorf(err, "attribute_indices"))
+	}
+	if err := checkValueType(prof.SampleType, dict); err != nil {
+		errs = errors.Join(errs, prefixErrorf(err, "sample_type"))
+	}
+	if err := checkValueType(prof.PeriodType, dict); err != nil {
+		errs = errors.Join(errs, prefixErrorf(err, "period_type"))
+	}
+	for i, s := range prof.Sample {
+		if err := checkSample(s, prof.TimeUnixNano, prof.TimeUnixNano+prof.DurationNano, dict); err != nil {
+			errs = errors.Join(errs, prefixErrorf(err, "sample[%d]", i))
+		}
+		// TODO: Check uniqueness of samples?
+		// Key: {stack_index, sorted(attribute_indices), link_index}
+		// Related: https://github.com/open-telemetry/opentelemetry-proto/issues/706.
+	}
+	for i, strIdx := range prof.CommentStrindices {
+		if err := checkIndex(len(dict.StringTable), strIdx); err != nil {
+			errs = errors.Join(errs, prefixErrorf(err, "comment_strindices[%d]", i))
+		}
+	}
```

Comment:
As the idea is to drop this field in the proto, we maybe can already drop the check here.
```suggestion
```